src/IrcConnection/IrcConnection.cs

```diff
@@ -1208,10 +895,21 @@ public WriteThread(IrcConnection connection)
             /// </summary>
             public void Start()
             {
-                _Thread = new Thread(new ThreadStart(_Worker));
-                _Thread.Name = "WriteThread ("+_Connection.Address+":"+_Connection.Port+")";
-                _Thread.IsBackground = true;
-                _Thread.Start();
+                _QueuedEvent.Reset();
+
+                ((Queue)_SendBuffer[Priority.High]).Clear();
+                ((Queue)_SendBuffer[Priority.AboveMedium]).Clear();
+                ((Queue)_SendBuffer[Priority.Medium]).Clear();
+                ((Queue)_SendBuffer[Priority.BelowMedium]).Clear();
+                ((Queue)_SendBuffer[Priority.Low]).Clear();
+
+                // Using lock for thread-safe access to _Thread
+                lock (this) {
```

Comment:
Why is this lock needed? API docs of Thread class says: "This type is thread safe." I would prefer a local var over locking "this"